ViewController.swift

```diff
@@ -6,14 +6,156 @@
 //
 
 import UIKit
-
-class ViewController: UIViewController {
-
+	
+class ViewController: UICollectionViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    var people: [Person] = []
+    
+    
+    let loadingActivityIndicator: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView()
+            indicator.style = .large
+            indicator.color = .white
+            indicator.startAnimating()
+            indicator.autoresizingMask = [.flexibleLeftMargin, .flexibleRightMargin, .flexibleTopMargin, .flexibleBottomMargin]
+            return indicator
+    }()
+    let blurEffectView: UIVisualEffectView = {
+        let blurEffect = UIBlurEffect(style: .dark)
+            let blurEffectView = UIVisualEffectView(effect: blurEffect)
+            blurEffectView.alpha = 0.8
+            blurEffectView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+            return blurEffectView
+        }()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(addNewPerson))
     }
 
-
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        people.count
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let person = people[indexPath.item]
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Person", for: indexPath) as? PersonCell, let path = getDocumentsDirectory()?.appendingPathComponent(person.image) else {
+            fatalError("Enable to deque PersonCell")
+        }
+        cell.name.text = person.name
+        cell.imageView.image = UIImage(contentsOfFile: path.path)
+        cell.imageView.layer.borderColor = UIColor(white: 0, alpha: 0.3).cgColor
+        cell.imageView.layer.borderWidth = 2
+        cell.imageView.layer.cornerRadius = 3
+        cell.layer.cornerRadius = 7
+        
+        return cell
+    }
+    
+    @objc private func addNewPerson() {
+        startLoadingAnimation()
+        let picker = UIImagePickerController()
+        picker.allowsEditing = true
+        picker.delegate = self
+        present(picker, animated: true)
```

Comment:
There is still some lag when hitting the plus button. This is happening because the tasks of showing the background and loading the image picker are happening in sequence. To fix that, you can schedule the image picker loading to run asynchronously. This is how you do it:

```suggestion
        DispatchQueue.main.async {
                let picker = UIImagePickerController()
                picker.allowsEditing = true
                picker.delegate = self
                present(picker, animated: true)
        }
```

Please notice the task is being schedule to run on the main thread (the one responsible for updating the UI). Not sure if this is part of Paul Hudson's 100 days of swift, but he has a nice article about `DispatchQueue`: https://www.hackingwithswift.com/read/9/3/gcd-101-async.